src/cozmo/pets.py

```diff
@@ -0,0 +1,250 @@
+# Copyright (c) 2016 Anki, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License in the file LICENSE.txt or at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+'''Pet recognition.
+
+Cozmo is capable of detecting pet faces (cats and dogs).
+
+The :class:`cozmo.world.World` object keeps track of pets the robot currently
+knows about, along with those that are currently visible to the camera.
+
+Each pet is assigned a :class:`Pet` object, which generates a number of
+observable events whenever the pet is observed, etc.
+
+If a pet goes off-screen, it will be assigned a new object_id (and
+therefore Pet object) when it returns. This is because the system can
+only tell if something appears to be a cat or a dog; it cannot recognise
+a specific pet, or e.g. tell the difference between two dogs.
+
+Note that these pet-specific events are also passed up to the
+:class:`cozmo.world.World` object, so events for all known pets can be
+observed by adding handlers there.
+'''
+
+# __all__ should order by constants, event classes, other classes, functions.
+__all__ = ['PET_VISIBILITY_TIMEOUT', 'PET_TYPE_UNKOWN', 'PET_TYPE_CAT', 'PET_TYPE_DOG',
+           'EvtPetAppeared', 'EvtPetDisappeared', 'EvtPetObserved',
+           'Pet']
+
+
+import math
+import time
+
+from . import logger
+
+from . import event
+from . import objects
+from . import util
+
+from ._clad import _clad_to_game_anki
+
+
+#: Length of time to go without receiving an observed event before
+#: assuming that Cozmo can no longer see a pet.
+PET_VISIBILITY_TIMEOUT = objects.OBJECT_VISIBILITY_TIMEOUT
+
+# Pet types that Cozmo can distinguish
+#: Pet Type reported by Cozmo when unsure of type of pet
+PET_TYPE_UNKOWN = "Unknown"
+#: Pet Type reported by Cozmo when he thinks it's a cat
+PET_TYPE_CAT = "Cat"
+#: Pet Type reported by Cozmo when he thinks it's a dog
+PET_TYPE_DOG = "Dog"
```

Comment:
nit pick, but can we make the constant values lowercase so there's one less thing to think about? e.g. `PET_TYPE_DOG = "dog"`